contracts/modules/capital/Pool.sol

```diff
@@ -0,0 +1,307 @@
+      require(balanceBefore < minAmount[toTokenAddress], 'balanceBefore >= min');
+      require(balanceBefore.add(amountOutMin) <= maxAmount[toTokenAddress], 'balanceAfter > max');
+    }
+
+    address[] memory path = new address[](2);
+    path[0] = router.WETH();
+    path[1] = toTokenAddress;
+    // TODO: pass deadline from off-chain?
+    router.swapExactETHForTokens.value(amountIn)(amountOutMin, path, address(this), block.timestamp);
+
+    uint balanceAfter = toToken.balanceOf(address(this));
+    uint amountOut = balanceAfter.sub(balanceBefore);
+    require(balanceAfter.sub(balanceBefore) >= amountOutMin, 'amount out too small');
+
+    lastSwapTime[toTokenAddress] = block.timestamp;
+
+    emit Swapped(ETH, toTokenAddress, amountIn, amountOut);
+  }
+
+  function swapTokensForETH(
+    address fromTokenAddress,
+    uint amountIn,
+    uint amountOutMin
+  ) external onlySwapController nonReentrant {
+
+    IERC20 fromToken = IERC20(fromTokenAddress);
+    uint balanceBefore = address(this).balance;
+
+    // TODO: should we allow more frequent trades?
+    uint timeSinceLastTrade = block.timestamp.sub(lastSwapTime[fromTokenAddress]);
+    require(timeSinceLastTrade > twapOracle.periodSize(), 'too fast');
+
+    {
+      // scope for liquidity check
+      address pairAddress = twapOracle.pairFor(fromTokenAddress, ETH);
+      IUniswapV2Pair pair = IUniswapV2Pair(pairAddress);
+      (uint112 reserve0, uint112 reserve1, /* time */) = pair.getReserves();
+
+      uint tokenReserve = fromTokenAddress < ETH ? reserve0 : reserve1;
+      uint maxTradable = tokenReserve.mul(MAX_TRADABLE_PAIR_LIQUIDITY).div(1e18);
+
+      require(amountIn <= maxTradable, 'exceeds max tradable amount');
+    }
+
+    {
+      // scope for token checks
+      uint avgAmountOut = twapOracle.consult(fromTokenAddress, amountIn, ETH);
+      uint maxSlippageAmount = avgAmountOut.mul(MAX_SLIPPAGE).div(1e18);
+      uint minOutOnMaxSlippage = avgAmountOut.sub(maxSlippageAmount);
+
+      require(amountOutMin > minOutOnMaxSlippage, 'max slippage exceeded');
+      require(balanceBefore > maxAmount[fromTokenAddress], 'balanceBefore <= max');
+      require(balanceBefore.sub(amountIn) >= minAmount[fromTokenAddress], 'balanceAfter < min');
+    }
+
+    address[] memory path = new address[](2);
+    path[0] = address(fromToken);
+    path[1] = router.WETH();
+    fromToken.safeApprove(address(router), amountIn);
+    // TODO: pass deadline from off-chain?
+    router.swapExactTokensForETH(amountIn, amountOutMin, path, address(this), block.timestamp);
+
+    uint balanceAfter = address(this).balance;
+    uint amountOut = balanceAfter.sub(balanceBefore);
+    require(amountOut >= amountOutMin, 'amount out too small');
+
+    emit Swapped(fromTokenAddress, ETH, amountIn, amountOut);
+  }
+
+  /* pool lifecycle functions */
+
+  function transferAsset(address asset, uint amount, address payable destination) external onlyGovernance nonReentrant {
```

Comment:
transferEther used to be in the previous verrsion of Pool1. Doesn't seem to be here. do we not have a use case for that anymore?